android/app/src/main/kotlin/coop/polypoly/polypod/polyOut/PolyOut.kt

```diff
@@ -103,29 +103,47 @@ open class PolyOut(
 
     open suspend fun readdir(
         id: String
-    ): Array<String> {
-
+    ): Array<Map<String, String>> {
         val fs = Preferences.getFileSystem(context)
+        val retList = mutableListOf(mapOf<String,String>())
+        
+        retList.removeFirst()
         if (id == "") {
             val newFs = fs.filter {
                 File(idToPath(it.key, context)).exists()
             }
             Preferences.setFileSystem(context, newFs)
-            return newFs.keys.toTypedArray()
+
+            val allIds = newFs.keys;
+            
+            for (idPath in allIds){
+                val relPath = getRelativePathFromId(idPath)
+                val idMap = mapOf<String,String>("id" to idPath, "path" to relPath)
+                retList.add(idMap)
+            }
+            return retList.toTypedArray()
         }
         if (readdirCache.contains(id)) {
             return readdirCache.get(id)!!
         }
-        val retList = mutableListOf<String>()
+
         File(idToPath(id, context)).walkTopDown().forEach {
-            retList.add(
-                "$fsFilesRoot/" + pathToId(it, context).removePrefix(fsPrefix)
-            )
+            val idValue = "$fsFilesRoot/" + pathToId(it, context).removePrefix(fsPrefix)
+            val relPath = getRelativePathFromId(idValue)
```

Comment:
Waaait a second -> We call a function called `pathToId` up there, with `it`. That means _we already have the path_, right? It's `it`. By my measure, `getRelativePathFromId` shouldn't be necessary at all.

Reply:
yyyesss I think that makes sense! In that case we indeed do not need `getRelativePathFromId `